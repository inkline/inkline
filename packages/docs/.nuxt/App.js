import Vue from 'vue'

import {
  getMatchedComponentsInstances,
  promisify,
  globalHandleError
} from './utils'

import NuxtLoading from './components/nuxt-loading.vue'

import '../css/index.scss'

import '../node_modules/@fortawesome/fontawesome-svg-core/styles.css'

import _6f6c098b from '../layouts/default.vue'
import _8f5e12f8 from '../layouts/documentation.vue'

const layouts = { "_default": _6f6c098b,"_documentation": _8f5e12f8 }

export default {
  head: {"title":"%s - Inkline","meta":[{"charset":"utf-8"},{"name":"viewport","content":"width=device-width, initial-scale=1, shrink-to-fit=no"},{"name":"keywords","content":"inkline, vue, vue 2, vue.js, vue components, framework, ui, ux"},{"hid":"description","name":"description","content":"Inkline is a modern UI\u002FUX framework for Vue.js, designed for creating flawless content-rich responsive web applications."},{"name":"msapplication-TileColor","content":"#ffffff"},{"name":"msapplication-TileImage","content":"\u002Ffavicon\u002Fms-icon-144x144.png"},{"name":"theme-color","content":"#ffffff"},{"hid":"og:site_name","property":"og:site_name","content":"Inkline"},{"hid":"og:type","property":"og:type","content":"website"},{"hid":"og:image","property":"og:image","content":"https:\u002F\u002Finkline.io\u002Fimages\u002Fpages\u002Findex.og.jpg"},{"hid":"mobile-web-app-capable","name":"mobile-web-app-capable","content":"yes"},{"hid":"apple-mobile-web-app-title","name":"apple-mobile-web-app-title","content":"Inkline"},{"hid":"author","name":"author","content":"Alex Grozav"},{"hid":"og:title","name":"og:title","property":"og:title","content":"Inkline"},{"hid":"og:description","name":"og:description","property":"og:description","content":"Inkline's documentation and homepage is built with Nuxt and publicly hosted on Netlify. Inkline is a modern Vue.js UI\u002FUX framework, written and maintained by @alexgrozav."}],"link":[{"rel":"apple-touch-icon","sizes":"57x57","href":"\u002Ffavicon\u002Fapple-icon-57x57.png"},{"rel":"apple-touch-icon","sizes":"60x60","href":"\u002Ffavicon\u002Fapple-icon-60x60.png"},{"rel":"apple-touch-icon","sizes":"72x72","href":"\u002Ffavicon\u002Fapple-icon-72x72.png"},{"rel":"apple-touch-icon","sizes":"76x76","href":"\u002Ffavicon\u002Fapple-icon-76x76.png"},{"rel":"apple-touch-icon","sizes":"114x114","href":"\u002Ffavicon\u002Fapple-icon-114x114.png"},{"rel":"apple-touch-icon","sizes":"120x120","href":"\u002Ffavicon\u002Fapple-icon-120x120.png"},{"rel":"apple-touch-icon","sizes":"144x144","href":"\u002Ffavicon\u002Fapple-icon-144x144.png"},{"rel":"apple-touch-icon","sizes":"152x152","href":"\u002Ffavicon\u002Fapple-icon-152x152.png"},{"rel":"apple-touch-icon","sizes":"180x180","href":"\u002Ffavicon\u002Fapple-icon-180x180.png"},{"rel":"icon","type":"image\u002Fpng","sizes":"192x192","href":"\u002Ffavicon\u002Fandroid-icon-192x192.png"},{"rel":"icon","type":"image\u002Fpng","sizes":"16x16","href":"\u002Ffavicon\u002Ffavicon-16x16.png"},{"rel":"icon","type":"image\u002Fpng","sizes":"32x32","href":"\u002Ffavicon\u002Ffavicon-32x32.png"},{"rel":"icon","type":"image\u002Fpng","sizes":"96x96","href":"\u002Ffavicon\u002Ffavicon-96x96.png"},{"rel":"icon","type":"image\u002Fx-icon","href":"\u002Ffavicon\u002Ffavicon.ico"},{"rel":"manifest","href":"\u002F_nuxt\u002Fmanifest.8f5593c4.json"}],"htmlAttrs":{"lang":"en"},"style":[],"script":[{"src":"\u002F\u002Fwww.googletagmanager.com\u002Fgtm.js?id=GTM-KD44VC3&l=dataLayer","async":true}],"noscript":[{"hid":"gtm-noscript","innerHTML":"\u003Ciframe src=\"\u002F\u002Fwww.googletagmanager.com\u002Fns.html?id=GTM-KD44VC3&l=dataLayer\" height=\"0\" width=\"0\" style=\"display:none;visibility:hidden\"\u003E\u003C\u002Fiframe\u003E","pbody":true}],"__dangerouslyDisableSanitizersByTagID":{"gtm-noscript":["innerHTML"]}},

  render (h, props) {
    const loadingEl = h('NuxtLoading', { ref: 'loading' })

    const layoutEl = h(this.layout || 'nuxt')
    const templateEl = h('div', {
      domProps: {
        id: '__layout'
      },
      key: this.layoutName
    }, [layoutEl])

    const transitionEl = h('transition', {
      props: {
        name: 'layout',
        mode: 'out-in'
      },
      on: {
        beforeEnter (el) {
          // Ensure to trigger scroll event after calling scrollBehavior
          window.$nuxt.$nextTick(() => {
            window.$nuxt.$emit('triggerScroll')
          })
        }
      }
    }, [templateEl])

    return h('div', {
      domProps: {
        id: '__nuxt'
      }
    }, [
      loadingEl,

      transitionEl
    ])
  },

  data: () => ({
    isOnline: true,

    layout: null,
    layoutName: ''
  }),

  beforeCreate () {
    Vue.util.defineReactive(this, 'nuxt', this.$options.nuxt)
  },
  created () {
    // Add this.$nuxt in child instances
    Vue.prototype.$nuxt = this
    // add to window so we can listen when ready
    if (process.client) {
      window.$nuxt = this

      this.refreshOnlineStatus()
      // Setup the listeners
      window.addEventListener('online', this.refreshOnlineStatus)
      window.addEventListener('offline', this.refreshOnlineStatus)
    }
    // Add $nuxt.error()
    this.error = this.nuxt.error
    // Add $nuxt.context
    this.context = this.$options.context
  },

  mounted () {
    this.$loading = this.$refs.loading
  },
  watch: {
    'nuxt.err': 'errorChanged'
  },

  computed: {
    isOffline () {
      return !this.isOnline
    }
  },

  methods: {
    refreshOnlineStatus () {
      if (process.client) {
        if (typeof window.navigator.onLine === 'undefined') {
          // If the browser doesn't support connection status reports
          // assume that we are online because most apps' only react
          // when they now that the connection has been interrupted
          this.isOnline = true
        } else {
          this.isOnline = window.navigator.onLine
        }
      }
    },

    async refresh () {
      const pages = getMatchedComponentsInstances(this.$route)

      if (!pages.length) {
        return
      }
      this.$loading.start()

      const promises = pages.map((page) => {
        const p = []

        if (page.$options.fetch) {
          p.push(promisify(page.$options.fetch, this.context))
        }

        if (page.$options.asyncData) {
          p.push(
            promisify(page.$options.asyncData, this.context)
              .then((newData) => {
                for (const key in newData) {
                  Vue.set(page.$data, key, newData[key])
                }
              })
          )
        }

        return Promise.all(p)
      })
      try {
        await Promise.all(promises)
      } catch (error) {
        this.$loading.fail()
        globalHandleError(error)
        this.error(error)
      }
      this.$loading.finish()
    },

    errorChanged () {
      if (this.nuxt.err && this.$loading) {
        if (this.$loading.fail) {
          this.$loading.fail()
        }
        if (this.$loading.finish) {
          this.$loading.finish()
        }
      }
    },

    setLayout (layout) {
      if (!layout || !layouts['_' + layout]) {
        layout = 'default'
      }
      this.layoutName = layout
      this.layout = layouts['_' + layout]
      return this.layout
    },
    loadLayout (layout) {
      if (!layout || !layouts['_' + layout]) {
        layout = 'default'
      }
      return Promise.resolve(layouts['_' + layout])
    }
  },

  components: {
    NuxtLoading
  }
}
