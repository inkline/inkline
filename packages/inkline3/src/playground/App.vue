<template>
    <div id="layout">
        <aside id="sidebar">
            <ul>
                <li>
                    <router-link :to="{ name: 'index' }">
                        Home
                    </router-link>
                </li>
                <li>
                    Core
                </li>
                <li>
                    <router-link :to="{ name: 'docs-core-grid' }">
                        Grid
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-core-layout' }">
                        Layout
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-core-typography' }">
                        Typography
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-core-images' }">
                        Images
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-core-tables' }">
                        Tables
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-core-code' }">
                        Code
                    </router-link>
                </li>
                <li>
                    Forms
                </li>
                <li>
                    <router-link :to="{ name: 'docs-forms-form' }">
                        Form
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-forms-input' }">
                        Input
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-forms-number-input' }">
                        Number Input
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-forms-textarea' }">
                        Textarea
                    </router-link>
                </li>
                <li>
                    Components
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-alert' }">
                        Alert
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-badge' }">
                        Badge
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-breadcrumb' }">
                        Breadcrumb
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-button' }">
                        Button
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-button-group' }">
                        Button Group
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-card' }">
                        Card
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-collapsible' }">
                        Collapsible
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-dropdown' }">
                        Dropdown
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-hamburger-menu' }">
                        Hamburger Menu
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-header' }">
                        Header
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-icon' }">
                        Icon
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-list-group' }">
                        List Group
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-loader' }">
                        Loader
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-media' }">
                        Media
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-modal' }">
                        Modal
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-nav' }">
                        Nav
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-navbar' }">
                        Navbar
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-pagination' }">
                        Pagination
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-popover' }">
                        Popover
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-components-tooltip' }">
                        Tooltip
                    </router-link>
                </li>
                <li>
                    Utilities
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-border' }">
                        Border
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-clearfix' }">
                        Clearfix
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-color' }">
                        Color
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-display' }">
                        Display
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-embed' }">
                        Embed
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-flex' }">
                        Flex
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-float' }">
                        Float
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-overflow' }">
                        Overflow
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-overlay' }">
                        Overlay
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-position' }">
                        Position
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-sizing' }">
                        Sizing
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-spacing' }">
                        Spacing
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-text' }">
                        Text
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-vertical-align' }">
                        Vertical Align
                    </router-link>
                </li>
                <li>
                    <router-link :to="{ name: 'docs-utilities-visibility' }">
                        Visibility
                    </router-link>
                </li>
                <li>
                    Color mode
                </li>
                <li>
                    <label>
                        <input v-model="colorMode" name="color-mode" type="radio" value="system" /> System
                    </label>
                    <br/>
                    <label>
                        <input v-model="colorMode" name="color-mode" type="radio" value="light" /> Light
                    </label>
                    <br/>
                    <label>
                        <input v-model="colorMode" name="color-mode" type="radio" value="dark" /> Dark
                    </label>
                </li>
            </ul>
        </aside>
        <main id="content">
            <router-view />
        </main>
    </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
    name: 'App',
    data() {
        return {
            colorMode: this.$inkline.options.colorMode
        };
    },
    watch: {
        colorMode(value) {
            this.$inkline.options.colorMode = value;
        }
    }
});
</script>

<style lang="scss">
    html,
    body,
    #app,
    #layout {
        height: 100%;
        width: 100%;
    }

    #layout {
        display: flex;
    }

    #sidebar {
        display: flex;
        height: 100%;
        overflow: auto;
        width: 280px;
        border-right: 1px solid var(--border-color-light);

        .inkline.-dark & {
            border-right: 1px solid var(--border-color-dark);
        }
    }

    #content {
        display: block;
        height: 100%;
        width: 100%;
        overflow: auto;
        padding: 2rem;
        box-sizing: border-box;

        > article {
            width: 100%;
            height: 100%;
            box-sizing: border-box;
        }
    }
</style>

