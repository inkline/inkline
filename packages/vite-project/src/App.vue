<template>
  <img alt="Vue logo" src="./assets/logo.png" />
  <HelloWorld msg="Hello Vue 3 + Vite" />
</template>

<script setup>
import HelloWorld from './components/HelloWorld.vue'
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>