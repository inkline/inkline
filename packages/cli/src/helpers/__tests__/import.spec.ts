import {
    addFieldToDefaultExport,
    addImport,
    importFromExists,
    insertImport,
    updateExistingImport
} from '../import';

const exampleSingleLineImportCode = `import { createApp } from 'vue';
import App from './App.vue';
import router from './router';
import store from './store';

createApp(App)
  .use(store)
  .use(router)
  .mount('#app');`;

const exampleMultiLineImportCode = `import { createApp } from 'vue';
import App from './App.vue';
import {
    IButton,
    ICheckbox,
    ILayout
} from '@inkline/inkline';

createApp(App)
  .mount('#app');`;

const exampleDefaultExportSingleLineCode = `import { defineConfig } from 'vite';

export default defineConfig({});`;

const exampleDefaultExportMultiLineCode = `import { defineConfig } from 'vite';

export default defineConfig({
    modules: ['example']
});`;

const exampleDefaultExportMultiLineCodeComplex = `import { defineConfig } from 'vite';

export default defineConfig({
    modules: [
        exampleA,
        exampleB,
        exampleC
    ]
});`;

describe('importFromExists', () => {
    it('should return true if the single line import exists', () => {
        const result = importFromExists(exampleSingleLineImportCode, {
            from: 'vue'
        });
        expect(result).toBe(true);
    });

    it('should return true if the multi line import exists', () => {
        const result = importFromExists(exampleMultiLineImportCode, {
            from: '@inkline/inkline'
        });
        expect(result).toBe(true);
    });

    it('should return false if the import does not exist', () => {
        const result = importFromExists(exampleSingleLineImportCode, {
            from: 'pinia'
        });
        expect(result).toBe(false);
    });
});

describe('insertImport', () => {
    it('should insert a default import', () => {
        const result = insertImport(exampleSingleLineImportCode, {
            name: 'Pinia',
            from: 'pinia'
        });

        expect(result).toContain(`import Pinia from 'pinia';`);
        expect(result).toMatchSnapshot();
    });

    it('should insert a named import', () => {
        const result = insertImport(exampleSingleLineImportCode, {
            name: ['createPinia'],
            from: 'pinia'
        });

        expect(result).toContain(`import { createPinia } from 'pinia';`);
        expect(result).toMatchSnapshot();
    });

    it('should insert multiple named imports', () => {
        const result = insertImport(exampleSingleLineImportCode, {
            name: ['createPinia', 'defineStore'],
            from: 'pinia'
        });

        expect(result).toContain(`import { createPinia, defineStore } from 'pinia';`);
        expect(result).toMatchSnapshot();
    });

    it('should insert a import when there are no other imports', () => {
        const result = insertImport(`const a = 1;`, {
            name: ['Inkline'],
            from: '@inkline/inkline'
        });

        expect(result).toContain(`import { Inkline } from '@inkline/inkline';`);
        expect(result).toMatchSnapshot();
    });
});

describe('updateExistingImport', () => {
    it('should update a single line named import', () => {
        const result = updateExistingImport(`import { createApp } from 'vue';`, {
            name: ['defineComponent'],
            from: 'vue'
        });

        expect(result).toEqual(`import { createApp, defineComponent } from 'vue';`);
    });

    it('should update a multi line named import', () => {
        const result = updateExistingImport(
            `import {\n  IButton,\n  ICheckbox,\n  ILayout\n} from '@inkline/inkline';`,
            {
                name: ['IColumn'],
                from: '@inkline/inkline'
            }
        );

        expect(result).toEqual(
            `import { IButton, ICheckbox, ILayout, IColumn } from '@inkline/inkline';`
        );
    });

    it('should update a default and named import', () => {
        const result = updateExistingImport(
            `import Inkline, {\n  IButton,\n  ICheckbox,\n  ILayout\n} from '@inkline/inkline';`,
            {
                name: ['IColumn'],
                from: '@inkline/inkline'
            }
        );

        expect(result).toEqual(
            `import Inkline, { IButton, ICheckbox, ILayout, IColumn } from '@inkline/inkline';`
        );
    });

    it('should add named imports to default import', () => {
        const result = updateExistingImport(`import Inkline from '@inkline/inkline';`, {
            name: ['IColumn'],
            from: '@inkline/inkline'
        });

        expect(result).toEqual(`import Inkline, { IColumn } from '@inkline/inkline';`);
    });

    it('should add default import to named imports', () => {
        const result = updateExistingImport(`import { IColumn } from '@inkline/inkline';`, {
            name: 'Inkline',
            from: '@inkline/inkline'
        });

        expect(result).toEqual(`import Inkline, { IColumn } from '@inkline/inkline';`);
    });
});

describe('addImport', () => {
    it('should insert a default import', () => {
        const result = addImport(exampleSingleLineImportCode, {
            name: 'Pinia',
            from: 'pinia'
        });

        expect(result).toContain(`import Pinia from 'pinia';`);
        expect(result).toMatchSnapshot();
    });

    it('should insert a named import', () => {
        const result = addImport(exampleSingleLineImportCode, {
            name: ['Inkline'],
            from: '@inkline/inkline'
        });

        expect(result).toContain(`import { Inkline } from '@inkline/inkline';`);
        expect(result).toMatchSnapshot();
    });

    it('should update a named single line import', () => {
        const result = addImport(exampleMultiLineImportCode, {
            name: ['defineComponent'],
            from: 'vue'
        });

        expect(result).toContain(`import { createApp, defineComponent } from 'vue';`);
        expect(result).toMatchSnapshot();
    });

    it('should update a named multi line import', () => {
        const result = addImport(exampleMultiLineImportCode, {
            name: ['IContainer'],
            from: '@inkline/inkline'
        });

        expect(result).toContain(
            `import { IButton, ICheckbox, ILayout, IContainer } from '@inkline/inkline';`
        );
        expect(result).toMatchSnapshot();
    });

    it('should add a default import to a named multi line import', () => {
        const result = addImport(exampleMultiLineImportCode, {
            name: 'Inkline',
            from: '@inkline/inkline'
        });

        expect(result).toContain(
            `import Inkline, { IButton, ICheckbox, ILayout } from '@inkline/inkline';`
        );
        expect(result).toMatchSnapshot();
    });
});

describe('addFieldToDefaultExport', () => {
    it('should add a field to a single-line default export', () => {
        const result = addFieldToDefaultExport(exampleDefaultExportSingleLineCode, 'plugins', [
            'Example'
        ]);

        expect(result).toContain(`plugins: [Example]`);
        expect(result).toMatchSnapshot();
    });

    it('should add a field to a multi-line default export', () => {
        const result = addFieldToDefaultExport(exampleDefaultExportMultiLineCode, 'plugins', [
            'Example'
        ]);

        console.log(result);

        expect(result).toContain(`plugins: [Example]`);
        expect(result).toMatchSnapshot();
    });

    it('should add a field to a multi-line default export with pre-existing value', () => {
        const result = addFieldToDefaultExport(exampleDefaultExportMultiLineCode, 'modules', [
            'Example'
        ]);

        expect(result).toContain(`modules: [Example, 'example']`);
        expect(result).toMatchSnapshot();
    });

    it('should add a field to a multi-line default export with pre-existing multi-line value', () => {
        const result = addFieldToDefaultExport(
            exampleDefaultExportMultiLineCodeComplex,
            'modules',
            ['Example']
        );

        console.log(result);

        expect(result).toContain(`modules: [Example,`);
        expect(result).toMatchSnapshot();
    });
});
