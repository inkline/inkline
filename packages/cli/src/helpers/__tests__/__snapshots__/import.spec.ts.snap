// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`addFieldToDefaultExport > should add a field to a multi-line default export 1`] = `
"import { defineConfig } from 'vite';

export default defineConfig({
    plugins: [Example],
    modules: ['example']
});"
`;

exports[`addFieldToDefaultExport > should add a field to a multi-line default export with pre-existing multi-line value 1`] = `
"import { defineConfig } from 'vite';

export default defineConfig({
    modules: [Example, 
        exampleA,
        exampleB,
        exampleC
    ]
});"
`;

exports[`addFieldToDefaultExport > should add a field to a multi-line default export with pre-existing value 1`] = `
"import { defineConfig } from 'vite';

export default defineConfig({
    modules: [Example, 'example']
});"
`;

exports[`addFieldToDefaultExport > should add a field to a single-line default export 1`] = `
"import { defineConfig } from 'vite';

export default defineConfig({
    plugins: [Example],
});"
`;

exports[`addImport > should add a default import to a named multi line import 1`] = `
"import { createApp } from 'vue';
import App from './App.vue';
import Inkline, { IButton, ICheckbox, ILayout } from '@inkline/inkline';

createApp(App)
  .mount('#app');"
`;

exports[`addImport > should insert a default import 1`] = `
"import { createApp } from 'vue';
import App from './App.vue';
import router from './router';
import store from './store';
import Pinia from 'pinia';

createApp(App)
  .use(store)
  .use(router)
  .mount('#app');"
`;

exports[`addImport > should insert a named import 1`] = `
"import { createApp } from 'vue';
import App from './App.vue';
import router from './router';
import store from './store';
import { Inkline } from '@inkline/inkline';

createApp(App)
  .use(store)
  .use(router)
  .mount('#app');"
`;

exports[`addImport > should update a named multi line import 1`] = `
"import { createApp } from 'vue';
import App from './App.vue';
import { IButton, ICheckbox, ILayout, IContainer } from '@inkline/inkline';

createApp(App)
  .mount('#app');"
`;

exports[`addImport > should update a named single line import 1`] = `
"import { createApp, defineComponent } from 'vue';
import App from './App.vue';
import {
    IButton,
    ICheckbox,
    ILayout
} from '@inkline/inkline';

createApp(App)
  .mount('#app');"
`;

exports[`insertImport > should insert a default import 1`] = `
"import { createApp } from 'vue';
import App from './App.vue';
import router from './router';
import store from './store';
import Pinia from 'pinia';

createApp(App)
  .use(store)
  .use(router)
  .mount('#app');"
`;

exports[`insertImport > should insert a import when there are no other imports 1`] = `
"import { Inkline } from '@inkline/inkline';

const a = 1;"
`;

exports[`insertImport > should insert a named import 1`] = `
"import { createApp } from 'vue';
import App from './App.vue';
import router from './router';
import store from './store';
import { createPinia } from 'pinia';

createApp(App)
  .use(store)
  .use(router)
  .mount('#app');"
`;

exports[`insertImport > should insert multiple named imports 1`] = `
"import { createApp } from 'vue';
import App from './App.vue';
import router from './router';
import store from './store';
import { createPinia, defineStore } from 'pinia';

createApp(App)
  .use(store)
  .use(router)
  .mount('#app');"
`;
